File: languages/java/oop/01_Student.java
/*
Objetivo: definir uma classe simples.
Como executar: javac 01_Student.java 02_StudentDemo.java && java StudentDemo
Conceitos treinados: atributos, construtor, método.
Próximo exercício: collections/01_ListExample.java
*/
class Student {
    private String name;
    private double grade;

    public Student(String name, double grade) {
        this.name = name;
        this.grade = grade;
    }

    public String getName() {
        return name;
    }

    public double getGrade() {
        return grade;
    }

    public String getStatus() {
        return grade >= 6.0 ? "Aprovado" : "Reprovado";
    }
}

File: languages/php/oop/01_student.php
<?php
/*
Objetivo: definir uma classe simples em PHP.
Como executar: php .\languages\php\oop\02_student_demo.php
Conceitos treinados: classe, construtor, método.
Próximo exercício: ../mini_projects/01_grade_report.php
*/

class Student {
    public string $name;
    public float $grade;

    public function __construct(string $name, float $grade) {
        $this->name = $name;
        $this->grade = $grade;
    }

    public function status(): string {
        return $this->grade >= 6.0 ? 'Aprovado' : 'Reprovado';
    }
}
